fyrox-ui/src/popup.rs

```diff
@@ -452,13 +466,27 @@ impl Control for Popup {
                             ));
                         }
                     }
+                    PopupMessage::Owner(owner) => {
+                        self.owner = *owner;
+                    }
+                    PopupMessage::Event(_) => (),
                 }
             }
         } else if let Some(WidgetMessage::KeyDown(key)) = message.data() {
             if !message.handled() && *key == KeyCode::Escape {
                 ui.send_message(PopupMessage::close(self.handle, MessageDirection::ToWidget));
                 message.set_handled(true);
             }
+        } else if self.owner.is_some()
```

Comment:
In retrospect, I don't know why I thought it might be a good idea to limit which messages were cloned. Any message might someday be useful to some widget. Maybe it should even include relaying already handled messages.